src/inmanta/agent/executor.py

```diff
@@ -123,7 +123,7 @@ def blueprint_hash(self) -> str:
         """
         if self._hash_cache is None:
             blueprint_dict: Dict[str, Any] = {
-                "pip_config": self.pip_config.dict(),
+                "pip_config": self.pip_config.model_dump(),
```

Comment:
I got deprecation warnings for these now, they should never have been there in the first place so I fixed it while I was at it.